Namespace(batch_size=4000, cpuonly=False, dataset='inaturalist', downloadall=False, end=10000, freeze=True, freeze_idx=36, model='vgg19', num_epochs=1, sequential=False, split_choice='automatic', split_idx=100, start=0, testonly=False, use_intermediate=False)
==> Preparing data..
==> Building model..
Using split index: 100
Freezing the lower layers of the model (vgg19) till index 36
The mode is:  vanilla
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Read 1056.3339109420776 MBs for this batch
Streaming inaturalist data took 21.501336574554443 seconds
The mode is:  vanilla

Epoch: 0
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Memory occpied: (1934.0, 3.0)
Time of next(dataloader) is: 17.44918155670166
Time for copying to cuda: 0.7431888580322266
Memory occpied: (4232.0, 1314.0)
Memory occpied: (4232.0, 1662.0)
Memory occpied: (4232.0, 2098.0)
Exception: Caught RuntimeError in replica 0 on device 0.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/parallel/parallel_apply.py", line 61, in _worker
    output = module(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torchvision/models/vgg.py", line 66, in forward
    x = self.features(x)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/container.py", line 141, in forward
    input = module(input)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/conv.py", line 447, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/conv.py", line 443, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 23.93 GiB (GPU 0; 14.76 GiB total capacity; 2.89 GiB already allocated; 10.62 GiB free; 2.91 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

Memory occpied: (4232.0, 3070.0)
Read 1077.8234357833862 MBs for this batch
Streaming inaturalist data took 24.416040658950806 seconds
The whole process took 53.70387387275696 seconds
